### PR TITLE
#46525 Addition of type param to refresh

### LINKED ISSUE #46525: Add type param to Refresh endpoint
## Description
This ticket should track the body of work to include the `type` query parameter on the Sign in Service (SiS) refresh endpoint.  This serves functionality to help keep track of the different service providers sessions.

## Tasks
- [ ] On the SiS (OAuth) include the `type` query parameter with the current authenticated service provider

## Acceptance Criteria
- [ ] The dev/staging/prod environments of DataDog should display the different types of refreshes.

### PR DESCRIPTION
**Pull request by acaruso-oddball**
## Description
Type param added to SiS refresh endpoint to allow for the tracking sessions created by different service providers.

## Original issue(s)
closes [Add type param to Refresh endpoint #46525](https://github.com/department-of-veterans-affairs/va.gov-team/issues/46525)

## Testing done
Unit

## Acceptance criteria
- [x]  On the SiS (OAuth) include the type query parameter with the current authenticated service provider
- [ ] The dev/staging/prod environments of DataDog should display the different types of refreshes.

## Definition of done
- [x] Events are logged appropriately
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: src/platform/utilities/oauth/utilities.js
Comment (AGarcia-Clarity) [inline src/platform/utilities/oauth/utilities.js:226]: Destructure this out in the parameters & change the `csp` to `type.

File: src/platform/user/authentication/components/SessionTimeoutModal.jsx
Comment (AGarcia-Clarity) [inline src/platform/user/authentication/components/SessionTimeoutModal.jsx:72]: 🚀

**Review by asg5704 — APPROVED** [review]

**Comment by acaruso-oddball** [comment]
Held up by track claims page cypress tests. They will possibly need to be fully refactored with [best practices](https://docs.cypress.io/guides/references/best-practices) in mind.

**Review by longmd — APPROVED** [review]
LGTM 🚀

**Review by asg5704 — APPROVED** [review]

**Comment by AGarcia-Clarity** [comment]
@department-of-veterans-affairs/vsp-release-tools @department-of-veterans-affairs/vsa-debt-frontend @department-of-veterans-affairs/va-cto-health-products 

Can you please review this thank you

**Comment by acaruso-oddball** [comment]
@department-of-veterans-affairs/vsp-release-tools 
@department-of-veterans-affairs/vsa-debt-frontend  
@department-of-veterans-affairs/va-cto-health-products 
@department-of-veterans-affairs/vsa-public-websites-frontend 
@department-of-veterans-affairs/benefits-team-1-frontend 

Hi folks, looking to get this merged, can someone from your team review this, thank you. I had to update some cypress tests by adding more explicit test names in order to have them pass. Feel free to reach out if you have questions.

**Review by rmessina1010 — APPROVED** [review]
LGTM

File: src/platform/utilities/oauth/utilities.js
Comment (ryguyk) [inline src/platform/utilities/oauth/utilities.js:219]: I'm curious if there's any particular reason why this is an object rather than just a single parameter named `type`.

**Review by ryguyk — APPROVED** [review]
Looks clean. I don't understand the context enough to know for sure if things are going to work, but the code is understandable. I did leave one question out of curiosity.

**Review by ScottyRJames — APPROVED** [review]

**Review by stephenBarrs — APPROVED** [review]
I will try to fix the linting issues in `coronavirus-vaccination` when I have time, but do not want to hold up this PR for those. I do not know enough overall context to know this PR will completely work as expected but at a high level it looks good to me.

File: src/platform/utilities/oauth/utilities.js
Comment (acaruso-oddball) [reply src/platform/utilities/oauth/utilities.js:219]: Hey Ryan, no particular reason for type to be destructured, was just helpful when I was adding new functionality. I appreciate the question and will make sure to keep it DRYer next time.

**Review by Dr-Pongo — APPROVED** [review]
LGTM! 🚀

**Review by data-doge — APPROVED** [review]

**Comment by acaruso-oddball** [comment]
Merging was still blocked due to community-care tests not passing though they were not initially required. Tests skipped in order to merge identity changes. Was not able to decipher where to make updates in community-care files as they are not identity related and weren't as straightforward as other non-identity tests updated.  @ryanshaw @vbahinwillit @simiadebowale please review and approve these changes.

File: src/applications/vaos/tests/e2e/community-care.cypress.spec.js
Comment (acaruso-oddball) [reply src/applications/vaos/tests/e2e/community-care.cypress.spec.js:430]: Sorry about that, since these tests have been written cypress has updated how they do things, which is causing the current naming convention for the "findby" methods to be outdated and throw errors. These errors were causing the test suite to fail and blocked this pr. The tests will possibly need to be fully refactored with the updated [best practices](https://docs.cypress.io/guides/references/best-practices) in mind. I was able to update some of the cypress tests for other teams but couldn't find the code to update for the community-care specs. I'll update my initial comment to give proper detail.

Comment (vbahinwillit) [reply src/applications/vaos/tests/e2e/community-care.cypress.spec.js:430]: Where can I find more information about the naming convention for 'findBy'?

Comment (acaruso-oddball) [reply src/applications/vaos/tests/e2e/community-care.cypress.spec.js:430]: Definitely. So to start, I misspoke, the get method has somehow changed, not findby (I'll update above). [I've linked the best practices here](https://docs.cypress.io/guides/references/best-practices). Sorry for the confusion, this is all related to the get method and specificity of current test selectors. 

Currently the error I receive when running the community care cypress tests are: 
Screen Shot 2022-10-05 at 12 14 40 PM

Stemming from selectors written in this way:
Screen Shot 2022-10-05 at 12 06 24 PM

Here are the best practices for specifying a selector in cypress. Currently the tests are written with unadvised naming convention of the selectors: 
Screen Shot 2022-10-05 at 11 59 06 AM

Here is an example of how I fixed this for another cypress file:
Screen Shot 2022-10-05 at 11 57 35 AM
Screen Shot 2022-10-05 at 11 58 33 AM

It's just adding a cypress test id rather than using the classname.

Comment (vbahinwillit) [reply src/applications/vaos/tests/e2e/community-care.cypress.spec.js:430]: Thanks for the information. Just curious but do you have the latest change from master merged into your branch? I just checked out the main branch and ran the test and it works.

Comment (acaruso-oddball) [reply src/applications/vaos/tests/e2e/community-care.cypress.spec.js:430]: I created this PR 13 days ago, and at the time, I did [pull from the main branch](https://github.com/department-of-veterans-affairs/vets-website/pull/22256/commits). At the time of creation, I noticed that there were cypress errors that caused one of the required test suites to fail - Cypress E2E Tests(4). I parsed through the code and was able to update those tests. Then, over time, I obtained approval from members of the appropriate teams.

This morning when I went to merge, I could not. The issue stemmed from Cypress Test Suite (10), a suite that had been failing but did not have a flag noting that it was required at the time. It apparently was because it blocked CI. When attempting to fix this test by adding a test-id I noticed that over 20 of the selectors in the file used the generic 'usa-button' classname instead of a test-id or more specific selector. As I am not a member of the VAOS team, locating the appropriate files to update proved difficult. And after some time, I chose to skip the test and leave a note.

I just created a new branch off of main and [drafted a pull request](https://github.com/department-of-veterans-affairs/vets-website/pull/22335) with my changes sans the updates to cypress tests outside of identity. [The test suite containing tests owned by your team still fails](https://github.com/department-of-veterans-affairs/vets-website/actions/runs/3192089191/jobs/5209299036). 

GitHub is requiring me to receive approval from three members of your team in order to merge my branch. The identity related code on this branch has been held up for quite some time and members of my team are waiting on it. If you'd like to shed insight on what needs to be done to get your tests to pass, please do, but until then can we skip them so that my team can continue our work?

**Review by vbahinwillit — APPROVED** [review]

File: src/applications/vaos/tests/e2e/community-care.cypress.spec.js
Comment (ryanshaw) [reply src/applications/vaos/tests/e2e/community-care.cypress.spec.js:430]: 👋 Hey @acaruso-oddball, I have an [open PR](https://github.com/department-of-veterans-affairs/vets-website/pull/22345) that fixes the failing VAOS tests.

Comment (ryanshaw) [reply src/applications/vaos/tests/e2e/community-care.cypress.spec.js:430]: The fix has now been merged into main.

Comment (acaruso-oddball) [reply src/applications/vaos/tests/e2e/community-care.cypress.spec.js:430]: Hey @ryanshaw thanks I see it and will pull those changes. Thanks to the both of you!

**Pull request merged**
